### PR TITLE
crimson: run in foreground if possible, silence warnings

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
```
10:55:56 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.143843/0.156261  => rejected
10:55:56 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 36.0/40.0  => rejected
10:55:56 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 12.5033/10.583  => accepted
10:55:56 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.432122/0.397305  => rejected
10:55:56 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.142986/0.157126  => rejected
10:55:56 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 36.0/40.0  => rejected
10:55:56 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 10.116/11.2694  => accepted
10:55:56 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.430997/0.395341  => rejected
10:55:56 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.815851/0.808925  => accepted
10:55:56 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 208.0/207.0  => accepted
10:55:56 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 0.0/0.0  => accepted
10:55:56 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.0763321/0.0753446  => accepted
10:55:56 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.793642/0.868457  => rejected
10:55:56 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 203.0/222.0  => rejected
10:55:56 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 0.0/0.0  => accepted
10:55:56 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.0775884/0.0708836  => rejected
10:55:56 - WARNING  - cbt      - 9 tests failed out of 16
```

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
```
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.154352/0.156261  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 39.0/40.0  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 10.0167/10.583  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.403527/0.397305  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.153446/0.157126  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 39.0/40.0  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 9.84886/11.2694  => accepted
11:58:59 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.405675/0.395341  => accepted
11:58:59 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.788317/0.808925  => accepted
11:58:59 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 201.0/207.0  => accepted
11:58:59 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 0.0/0.0  => accepted
11:58:59 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.075925/0.0753446  => accepted
11:58:59 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.776106/0.868457  => rejected
11:58:59 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 198.0/222.0  => rejected
11:58:59 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 0.0/0.0  => accepted
11:58:59 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.0791125/0.0708836  => rejected
11:58:59 - WARNING  - cbt      - 3 tests failed out of 16
```

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

File: src/crimson/osd/pg.cc [hunk 1]
```diff
@@ -384,13 +384,13 @@ seastar::future<Ref<MOSDOpReply>> PG::do_osd_ops(Ref<MOSDOp> m)
...
     logger().debug("got ceph::osd::error while handling object {}: {} ({})",
                    oid, e.code(), e.what());
-
-    backend->evict_object_state(oid);
-    auto reply = make_message<MOSDOpReply>(
-      m.get(), -e.code().value(), get_osdmap_epoch(), 0, false);
-    reply->set_enoent_reply_versions(peering_state.get_info().last_update,
-                                       peering_state.get_info().last_user_version);
-    return seastar::make_ready_future<Ref<MOSDOpReply>>(std::move(reply));
+    return backend->evict_object_state(oid).then([=] {
```
Comment (athanatos) [inline src/crimson/osd/pg.cc]: If I'm understanding this correctly, we'll eventually want to do evict_object_state in the background, but we'll have to establish some kind of async work machinery with throttles.

**Pull request merged**